Add option nu to the Matern geometry and add tests

amal-ghamdi opened the pull request:
closes #45 
closes #49

babakmaboudi commented:
Looks great. Thank you Amal. My only comment is to set the default value of nu to 0.5 (which is mostly used).

nabriis commented on cuqipy_fenics/geometry.py:
```diff
@@ -140,10 +140,14 @@ class MaternExpansion(_WrappedGeometry):
-        Number of expantion terms to represent the Matern field realization
+        Number of expansion terms to represent the Matern field realization
+
+    nu : float 
```
Remember to update the docstring with default smoothness parameter.

nabriis commented on cuqipy_fenics/geometry.py:
```diff
@@ -177,12 +181,16 @@ class MaternExpansion(_WrappedGeometry):
+            raise ValueError("Smoothness parameter nu must be positive")
+            
         super().__init__(geometry)
         if not hasattr(geometry, 'mesh'):
```
Can you add a check here so that this geometry only works for FEniCS geoms. For now at least .)

amal-ghamdi replied:
done 👍

nabriis commented on cuqipy_fenics/geometry.py:
```diff
@@ -245,8 +257,10 @@ def par2field(self, p):
+        d = self.physical_dim
         for idx in range(Ns):
-            field_list[:,idx] = self.eig_vec@( np.sqrt(self.eig_val)*p[...,idx] )
+            field_list[:,idx] = self.eig_vec@( self.eig_val**((nu+d/2)/2)*p[...,idx] )
```
Could ref where this eq. is found. Perhaps in docstring?

amal-ghamdi replied:
@nabriis I added a particular reference.
@babakmaboudi, can you take a quick look at the reference I added, section 4.3, in particular last equation in the section. Thanks!

nabriis reviewed (approved):
Looks good to me. Minor suggestions on documentation, after incorporated you can merge

amal-ghamdi commented:
@nabriis and @babakmaboudi, thank you for your review! I made the updates and I requested re-review.

@nabriis, I incorporated your three comments. Plus I also changed the `MaternExpansion` name to `MaternKLExpansion` to match what we decided on in the paper.

@babakmaboudi I updated the default value. And would it possible that you check that the reference I added makes sense. Particularly last equation in page 11, section 4.3

Thank you both!

nabriis reviewed (approved):
LGTM

jakobsj reviewed (approved):
LGTM, go ahead
